codec-classes-quic/src/main/java/io/netty/incubator/codec/quic/BoringSSLCertificateCallback.java

```diff
@@ -77,12 +77,12 @@ final class BoringSSLCertificateCallback {
         KEY_TYPES.put("DH_RSA", KEY_TYPE_DH_RSA);
     }
 
-    private static final Set<String> SUPPORTED_KEY_TYPES = Collections.unmodifiableSet(new LinkedHashSet<>(
+    public static final Set<String> SUPPORTED_KEY_TYPES = new LinkedHashSet<>(
```

Comment:
why this change ?

Reply:
This allows the person implementing this quic library to use alternate x509 certificate types if they need too. With the change, they can now override the defaults hard-coded in BoringSSLCertificateCallback. 

This change goes along with the SSL_CTX_set1_sigalgs option. The SSL_CTX_set1_sigalgs config allows the user to use different signing algorithms, and this change allows them to change the "allowed" certificate types.

Reply:
I think we should just do this via the constructor and so via a `BoringSslContextOption`

Reply:
Reverted this change